lib/refile/backend/file_system.rb

```diff
@@ -34,7 +34,7 @@ def initialize(directory, max_size: nil, hasher: Refile::RandomHasher.new)
       # @return [Refile::File]      The uploaded file
       verify_uploadable def upload(uploadable)
         id = @hasher.hash(uploadable)
-        IO.copy_stream(uploadable, path(id))
+        ::File.open(path(id), "wb") { |dest| IO.copy_stream(uploadable, dest) }
```

Comment:
We found a solution like this earlier through trial and error, but the problem is that this is a performance regression. When given a path, IO.copy_stream will do a straight file copy via syscalls, which is much faster than buffering the file through Ruby. I'd like to preserve that behaviour if at all possible.